config/default.yml

```diff
@@ -1,6 +1,3 @@
-base_url: /
-static_url: /static
-express_port: 9001
 api_base: https://api.hel.fi/linkedevents-test/v1
```

Comment:
Is default.yml actually needed at this point?

Reply:
Yes it is. It is still used by the dev codepath.

Reply:
More to the point: why not change everything to use nconf?

Whatever goes reading default.yml didn't show up with grep nor WebStorm search so it must be very well hidden indeed...

Reply:
I'd certainly like to do that. Would be nice to have comments from @Rikuoja on how he prefers to inject configuration during development. I created a couple of scriptlets settings environment variables and sourced those.
But perhaps a file somewhere should be read if NODE_ENV === 'development'?

And default.yml is read by config.js. It is wonderfully inquisitive, as evidenced here: https://github.com/lorenwest/node-config/wiki/Configuration-Files

Reply:
Sigh. I didn't realize importing config module produces config object made by said module from file names built across many lines, hence the failure with string based searches...

Anyway, I'd very much like that there would be only *one* configuration system in effect.

Reply:
For me, the simplest way (that I do use in most cases) is to have a local file not included in the repo that contains dev settings.

Also, example dev settings need to be provided for external developers, so that they can get started with development with the minimum of fuss. In my view, a file existing in the repo is an easier way to do it than providing a README that specifies which environment variables must be set first.For me, the simplest way (that I do use in most cases) is to have a local file not included in the repo that contains dev settings.

Also, example dev settings need to be provided for external developers, so that they can get started with development with the minimum of fuss. In my view, a file existing in the repo is an easier way to do it than providing a README that specifies which environment variables must be set first.